### ISSUE TITLE
Newbie unable to bundle install

### ISSUE THREAD

**Issue by Nickk4** [refinery/refinerycms#2863]
I'm trying to create the rickrockstar project as described on http://refinerycms.com/guides/getting-started.

I have Rails 4.2 and 3.10 installed and understand that I then should run 3.10 because RefineryCMS does not support 4.2. So in the terminal I run: 
```rails _3.2.10 _ new rickrockstar -m http://refinerycms.com/t/2.1.0```

After creating a lot of files and folders I get the error message below. Can anyone suggest what to do?

Result:
```
Fetching gem metadata from https://rubygems.org/.........
Fetching additional metadata from https://rubygems.org/..
Resolving dependencies...
Bundler could not find compatible versions for gem "actionpack":
  In Gemfile:
    refinerycms-acts-as-indexed (~> 1.0.0) x86-mingw32 depends on
      refinerycms-core (~> 2.1.0) x86-mingw32 depends on
actionpack (!= 3.2.0, != 3.2.1, != 3.2.10, != 3.2.11, != 3.2.2, !=
3.2.3, != 3.2.4, != 3.2.5, != 3.2.6, != 3.2.7, != 3.2.8, != 3.2.9, < 3.3, >=
3.1.11) x86-mingw32

    rails (= 3.2.10) x86-mingw32 depends on
      actionmailer (= 3.2.10) x86-mingw32 depends on
        actionpack (3.2.10)
        rake    db:create
←[31mCould not find gem 'refinerycms-acts-as-indexed (~> 1.0.0) x86-mingw32' in
the gems available on this machine.←[0m
←[33mRun `bundle install` to install missing gems.←[0m
    generate    refinery:cms --fresh-installation rickrooooockstar -m http://ref
inerycms.com/t/2.1.0
←[31mCould not find gem 'refinerycms-acts-as-indexed (~> 1.0.0) x86-mingw32' in
the gems available on this machine.←[0m
←[33mRun `bundle install` to install missing gems.←[0m
    ============================================================================

    Your new Refinery CMS application is now installed and mounts at '/'
  ============================================================================
         run  bundle install
Fetching gem metadata from https://rubygems.org/.........
Fetching additional metadata from https://rubygems.org/..
Resolving dependencies...
Bundler could not find compatible versions for gem "actionpack":
  In Gemfile:
    refinerycms-acts-as-indexed (~> 1.0.0) x86-mingw32 depends on
      refinerycms-core (~> 2.1.0) x86-mingw32 depends on
actionpack (!= 3.2.0, != 3.2.1, != 3.2.10, != 3.2.11, != 3.2.2, !=
3.2.3, != 3.2.4, != 3.2.5, != 3.2.6, != 3.2.7, != 3.2.8, != 3.2.9, < 3.3, >=
3.1.11) x86-mingw32

    rails (= 3.2.10) x86-mingw32 depends on
      actionmailer (= 3.2.10) x86-mingw32 depends on
        actionpack (3.2.10)
```

**Comment by simi** [comment]
Can you try with Rails `3.2.12`?

```
gem install rails --version 3.2.12
rails _3.2.12 _ new rickrockstar -m http://refinerycms.com/t/2.1.0
```

**Comment by Nickk4** [comment]
Succesfully installed Rails 3.2.12.
Then I ran: rails _3.2.12_ new rickrockstar -m http://refinerycms.com/t/2.1.0
And got the result:

...
Installing awesome_nested_set 2.1.6

Gem::InstallError: babosa requires Ruby version >= 2.0.0.
An error occurred while installing babosa (1.0.2), and Bundler cannot continue.
Make sure that `gem install babosa -v '1.0.2'` succeeds before bundling.
        rake    db:create
←[31mCould not find gem 'refinerycms-acts-as-indexed (~> 1.0.0) x86-mingw32' in
the gems available on this machine.←[0m
←[33mRun `bundle install` to install missing gems.←[0m
    generate    refinery:cms --fresh-installation rickrockstar -m http://refiner
ycms.com/t/2.1.0
←[31mCould not find gem 'refinerycms-acts-as-indexed (~> 1.0.0) x86-mingw32' in
the gems available on this machine.←[0m
←[33mRun `bundle install` to install missing gems.←[0m
    ============================================================================

    Your new Refinery CMS application is now installed and mounts at '/'
  ============================================================================
         run  bundle install
Fetching gem metadata from https://rubygems.org/.........
Fetching additional metadata from https://rubygems.org/..
Resolving dependencies...
Using rake 10.4.2
Using i18n 0.7.0
Using multi_json 1.10.1
Using activesupport 3.2.12
Using builder 3.0.4
Using activemodel 3.2.12
Using erubis 2.7.0
Using journey 1.0.4
Using rack 1.4.5
Using rack-cache 1.2
Using rack-test 0.6.3
Using hike 1.2.3
Using tilt 1.4.1
Using sprockets 2.2.3
Using actionpack 3.2.12
Using mime-types 1.25.1
Using polyglot 0.3.5
Using treetop 1.4.15
Using mail 2.4.4
Using actionmailer 3.2.12
Using arel 3.0.3
Using tzinfo 0.3.42
Using activerecord 3.2.12
Using activeresource 3.2.12
Using acts_as_indexed 0.8.3
Using awesome_nested_set 2.1.6

Gem::InstallError: babosa requires Ruby version >= 2.0.0.
An error occurred while installing babosa (1.0.2), and Bundler cannot continue.
Make sure that `gem install babosa -v '1.0.2'` succeeds before bundling.

**Comment by ghoppe** [comment]
As the error says, the latest version of the babosa gem doesn't support Ruby less than 2.0.0. RefineryCMS doesn't specify which version of babosa to use so bundler tries to use the latest and fails. 

Either install Ruby 2.0+ and use that or add the following to your Gemfile
`gem 'babosa', '0.3.11'`

Run the `bundle update` command and try that.

**Comment by Nickk4** [comment]
Thanks, I thought I would take this as an opportunity to update rails. I installed it and again ran:
 rails _3.2.12_ new rickrockstar -m http://refinerycms.com/t/2.1.0
However, something seems to go wrong with the new ruby as it immediately results in:

D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/2.1.0/rubygems/dependency.rb:298:in `to
_specs': Could not find 'railties' (= 3.2.12) - did find: [railties-4.1.8] (Gem:
:LoadError)
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/2.1.0/rubygems/dependency.
rb:309:in `to_spec'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/2.1.0/rubygems/core_ext/ke
rnel_gem.rb:53:in `gem'
        from D:/row/RailsInstaller/Ruby2.1.0/bin/rails:22:in `<main>'

**Comment by Nickk4** [comment]
I'm a bit lost. I rebooted, which I hadn't done after installing Ruby 2.0+. Now the result is different but puzzling. I run rails _3.2.12_ new rickrockstar -m http://refinerycms.com/t/2.1.0 but this does not create anything and just gives as a result:

==================================
DL is deprecated, please use Fiddle
Usage:
  rails new APP_PATH [options]

Options:
  -r, [--ruby=PATH]                                      # Path to the Ruby bina
ry of your choice
                                                         # Default: D:/row/Rails
Installer/Ruby2.1.0/bin/ruby.exe
  -m, [--template=TEMPLATE]                              # Path to some applicat
ion template (can be a filesystem path or URL)
      [--skip-gemfile], [--no-skip-gemfile]              # Don't create a Gemfil
e
  -B, [--skip-bundle], [--no-skip-bundle]                # Don't run bundle inst
all
  -G, [--skip-git], [--no-skip-git]                      # Skip .gitignore file
      [--skip-keeps], [--no-skip-keeps]                  # Skip source control .
keep files
  -O, [--skip-active-record], [--no-skip-active-record]  # Skip Active Record fi
les
  -V, [--skip-action-view], [--no-skip-action-view]      # Skip Action View file
s
  -S, [--skip-sprockets], [--no-skip-sprockets]          # Skip Sprockets files
      [--skip-spring], [--no-skip-spring]                # Don't install Spring
application preloader
  -d, [--database=DATABASE]                              # Preconfigure for sele
cted database (options: mysql/oracle/postgresql/sqlite3/frontbase/ibm_db/sqlserv
er/jdbcmysql/jdbcsqlite3/jdbcpostgresql/jdbc)
                                                         # Default: sqlite3
  -j, [--javascript=JAVASCRIPT]                          # Preconfigure for sele
cted JavaScript library
                                                         # Default: jquery
  -J, [--skip-javascript], [--no-skip-javascript]        # Skip JavaScript files

      [--dev], [--no-dev]                                # Setup the application
 with Gemfile pointing to your Rails checkout
      [--edge], [--no-edge]                              # Setup the application
 with Gemfile pointing to Rails repository
  -T, [--skip-test-unit], [--no-skip-test-unit]          # Skip Test::Unit files

      [--rc=RC]                                          # Path to file containi
ng extra configuration options for rails command
      [--no-rc], [--no-no-rc]                            # Skip loading of extra
 configuration options from .railsrc file

Runtime options:
  -f, [--force]                    # Overwrite files that already exist
  -p, [--pretend], [--no-pretend]  # Run but do not make any changes
  -q, [--quiet], [--no-quiet]      # Suppress status output
  -s, [--skip], [--no-skip]        # Skip files that already exist

Rails options:
  -h, [--help], [--no-help]        # Show this help message and quit
  -v, [--version], [--no-version]  # Show Rails version number and quit

Description:
    The 'rails new' command creates a new Rails application with a default
    directory structure and configuration at the path you specify.

    You can specify extra command-line arguments to be used every time
    'rails new' runs in the .railsrc configuration file in your home directory.

    Note that the arguments specified in the .railsrc file don't affect the
    defaults values shown above in this help message.

Example:
    rails new ~/Code/Ruby/weblog

    This generates a skeletal Rails installation in ~/Code/Ruby/weblog.
    See the README in the newly created application to get going.
==================================

**Comment by bricesanchez** [comment]
Do you have install execjs ?

```bash
bundle install execjs
```

**Comment by Nickk4** [comment]
I'm not sure if I have that but when I run
bundle install execjs
I get the result:
=============================
DL is deprecated, please use Fiddle
ERROR: "bundle install" was called with arguments ["execjs"]
Usage: "bundle install [OPTIONS]"

**Comment by Nickk4** [comment]
P.S. How do you set the grey box around the code in your message here?

**Comment by bricesanchez** [comment]
https://guides.github.com/features/mastering-markdown/#examples

DL is deprecated, please use Fiddle : http://stackoverflow.com/questions/15590450/ruby-2-0-0p0-irb-error-dl-is-deprecated-please-use-fiddle

Do you try to install refinerycms on a windows ?

**Comment by Nickk4** [comment]
Yes, I'm working from a windows machine. 
And yes I've been trying to install refinerycms. which is why I included '''-m http://refinerycms.com/t/2.1.0''' in the rails new command. But also without refinerycms I get the error. So also when I run '''rails _3.2.12_ new rickrockstar''' withou the -m I get the result:
'''D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/2.1.0/rubygems/dependency.rb:298:in `to
_specs': Could not find 'railties' (= 3.2.12) - did find: [railties-4.1.8] (Gem:
:LoadError)
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/2.1.0/rubygems/dependency.
rb:309:in `to_spec'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/2.1.0/rubygems/core_ext/ke
rnel_gem.rb:53:in `gem'
        from D:/row/RailsInstaller/Ruby2.1.0/bin/rails:22:in `<main>'
'''

**Comment by Nickk4** [comment]
I've been able to solve some of the problem using the instructions from https://gist.github.com/fnichol/867550. I can now succesfulle create a project by running ```rails new proj3```.

I have also succesfully ran ```gem update --system``` to update ruby gems.

But still when I run ```rails _3.2.12_ new rickrockstar -m http://refinerycms.com/t/2.1.0``` the result is:
```
D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/site_ruby/2.1.0/rubygems/dependency.rb:
315:in `to_specs': Could not find 'railties' (= 3.2.12) - did find: [railties-4.
1.8] (Gem::LoadError)
Checked in 'GEM_PATH=C:/Users/Username/.gem/ruby/2.1.0;D:/row/RailsInstaller/Ruby2.
1.0/lib/ruby/gems/2.1.0', execute `gem env` for more information
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/site_ruby/2.1.0/rubygems/d
ependency.rb:324:in `to_spec'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/site_ruby/2.1.0/rubygems/c
ore_ext/kernel_gem.rb:64:in `gem'
        from D:/row/RailsInstaller/Ruby2.1.0/bin/rails:22:in `<main>'
```

**Comment by Nickk4** [comment]
I think what the error message is saying that rails 3.2.12 doesn't exist. Its installation may have been overwritten somewhere along the line? So I reinstalled it again using ```gem install rails --version 3.2.12```.
During its installation I got the error messages:
```
**... a lot of good things...**
Parsing documentation for builder-3.0.4
D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/rdoc-4.2.0/lib/rdoc/par
ser.rb:87: warning: Unsupported encoding : ignored
D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/rdoc-4.2.0/lib/rdoc/par
ser.rb:87: warning: Unsupported encoding  ignored
D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/rdoc-4.2.0/lib/rdoc/par
ser.rb:87: warning: Unsupported encoding : ignored
D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/rdoc-4.2.0/lib/rdoc/par
ser.rb:87: warning: Unsupported encoding  ignored
Installing ri documentation for builder-3.0.4
Parsing documentation for activemodel-3.2.12
Installing ri documentation for activemodel-3.2.12
**...more good things...**
Done installing documentation for activesupport, builder, activemodel, rack, rac
k-cache, journey, sprockets, actionpack, arel, tzinfo, activerecord, activeresou
rce, mime-types, polyglot, treetop, mail, actionmailer, rack-ssl, rdoc, railties
, rails after 149 seconds
21 gems installed
```
So although it did give warnings, it seems to have installed it.
After this I again tried  ```rails _3.2.12_ new rickrockstar -m http://refinerycms.com/t/2.1.0``` with as the result:
```
**...A lot of create, using and installing...**
        rake    db:create
    generate    refinery:cms --fresh-installation rickrockstar -m http://refiner
ycms.com/t/2.1.0
DL is deprecated, please use Fiddle
D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3.2.12/lib/rai
ls/railtie/configuration.rb:85:in `method_missing': undefined method `eager_load
' for #<Rails::Application::Configuration:0x45048c0> (NoMethodError)
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/decorators
-1.0.4/lib/decorators/railtie.rb:9:in `block in <class:Railtie>'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/lazy_load_hooks.rb:34:in `call'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/lazy_load_hooks.rb:34:in `execute_hook'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/lazy_load_hooks.rb:43:in `block in run_load_hooks'

        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/lazy_load_hooks.rb:42:in `each'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/lazy_load_hooks.rb:42:in `run_load_hooks'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/application/bootstrap.rb:69:in `block in <module:Bootstrap>'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/initializable.rb:30:in `instance_exec'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/initializable.rb:30:in `run'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/initializable.rb:55:in `block in run_initializers'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/initializable.rb:54:in `each'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/initializable.rb:54:in `run_initializers'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/application.rb:136:in `initialize!'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/railtie/configurable.rb:30:in `method_missing'
        from D:/row/dev/rickrockstar/config/environment.rb:5:in `<top (required)
>'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/dependencies.rb:251:in `require'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/dependencies.rb:251:in `block in require'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/dependencies.rb:236:in `load_dependency'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/activesupp
ort-3.2.12/lib/active_support/dependencies.rb:251:in `require'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/application.rb:103:in `require_environment!'
        from D:/row/RailsInstaller/Ruby2.1.0/lib/ruby/gems/2.1.0/gems/railties-3
.2.12/lib/rails/commands.rb:25:in `<top (required)>'
        from script/rails:6:in `require'
        from script/rails:6:in `<main>'
    ============================================================================

    Your new Refinery CMS application is now installed and mounts at '/'
  ============================================================================
         run  bundle install

**...A lot of using...**
Your bundle is complete!
Use `bundle show [gemname]` to see where a bundled gem is installed.
```

I'm not sure what it is that still goes wrong. Any ideas??

**Comment by parndt** [comment]
@Nickk4 my apologies, I caused this with a dependency but it's solved now with parndt/decorators#11
